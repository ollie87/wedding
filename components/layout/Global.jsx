
import { createGlobalStyle } from 'styled-components'

export const GlobalStyle = createGlobalStyle`
    html {
        font-size: 62.5%;
        box-sizing: border-box;
    }
    *, *:before, *:after {
        box-sizing: inherit;
    }
    body {
        font-size: 1.6rem;
        line-height: 1.5;
        font-family: 'Casta-Thin', serif;
    }
    h1, h2, h3 {
        margin: 0 0 2rem 0;
        line-height: 1.5;
    }
    h1, h2 {
        font-family: 'Donitta', serif;
        font-weight: 400;
        color:  ${({ theme: { colors } }) => colors.grey_darkest};
    }
    label {
        font-family: 'Casta-Thin', serif;
        font-weight: 800;
        font-size: 20px;
    }
    h3 {
        font-family: 'PT Sans', serif;
        color:  ${({ theme: { colors } }) => colors.grey_medium};
    }
    ul {
        list-style: none;
        margin: 0;
        padding: 0;
    }
    a {
        text-decoration: none;
    }
    p {
        color:  ${({ theme: { colors } }) => colors.grey_dark};
    }
    input, select, textarea {
        font-family: 'Casta-Thin', serif;
    }
`