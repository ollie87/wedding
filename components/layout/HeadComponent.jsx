import Head from 'next/head'

const HeadComponent = () => {
    return (
        <Head>
            <title>Boda Luc√≠a y Mario</title>
            <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/normalize/8.0.1/normalize.min.css" integrity="sha512-NhSC1YmyruXifcj/KFRWoC561YpHpc5Jtzgvbuzx5VozKpWvQ+4nXhPdFgmx8xqexRcpAglTj9sIBWINXa8x5w==" crossOrigin="anonymous" referrerpolicy="no-referrer" />
            <link rel="preconnect" href="https://fonts.googleapis.com" />
            <link rel="preconnect" href="https://fonts.gstatic.com" crossOrigin />
            <link href="https://fonts.googleapis.com/css2?family=PT+Sans:wght@400;700&family=Roboto+Slab:wght@400;700&display=swap" rel="stylesheet" />
        </Head>
    )
}

export default HeadComponent