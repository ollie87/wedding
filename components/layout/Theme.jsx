export const theme = {
    colors: {
        grey_darkest: '#333333',
        grey_dark: '#5C4E3A',
        grey_high: '#54504C',
        grey_medium: '#AFA8A3',
        grey_soft: '#DEDEDE',
        grey_light: '#EFEFEF',
        primary: 'red',
    }
};