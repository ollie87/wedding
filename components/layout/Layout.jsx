import React from 'react'
import { ThemeProvider } from 'styled-components'
import { theme } from './Theme'
import HeadComponent from './HeadComponent'
import { GlobalStyle } from './Global'

const Layout = props => {
    return (
        <>
            <HeadComponent />
            <GlobalStyle theme={theme} />
            <ThemeProvider theme={theme}>
                <main className='lg:grid lg:grid-flow-row-dense lg:grid-cols-1 lg:grid-rows-1'>
                    <div className='lg:container lg:mx-auto lg:justify-center'>
                        {props.children}
                    </div>
                </main>
            </ThemeProvider>
        </>
    )
}

export default Layout