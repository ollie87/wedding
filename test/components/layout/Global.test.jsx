import { render, screen } from '@testing-library/react'
import { GlobalStyle } from '../../../components/layout/Global';

describe('Global', () => {
    it("should have global style", () => {
        render(<GlobalStyle />);
        expect(document.head).toMatchSnapshot();
    });
})