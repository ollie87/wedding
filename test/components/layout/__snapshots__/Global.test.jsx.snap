// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Global should have global style 1`] = `<head />`;
